dpytools/validation/json/validation.py

```diff
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+from urllib.parse import urlparse
+
+import jsonschema
+from jsonschema import ValidationError
+
+"""
+# data_dict is for passing in a dictionary to be validated
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary)
+
+# msg is to include a helpful context when debugging (i.e "what we're validating")
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, msg="Some helpful message should this validation fail")
+
+validate_json_schema("/path/to/schema.json", data_path="/path/to/some/json", msg="Some helpful message should this validation fail")
+
+# indent should pretty print the json contents of the error to make it
+# more easily parsed by humans
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, indent=2)
+"""
+
+
+def validate_json_schema(
+    schema_path: str,
+    data_dict_or_path: Union[Dict, str],
+    msg: str = "",
+    indent: int = 2,
+) -> Optional[ValidationError]:
+    """
+    Validate metadata.json files against schema provided.
+
+    `schema_path`: file path of schema to validate against
+    `data_dict_or_path`: file path or dictionary of data to be validated
+    `msg`: optional string to provide additional information about validation
+    `indent`: optional integer to be used when indenting json output
+    """
+    # Load schema as dict
+    parsed_schema_path = urlparse(schema_path)
+    if parsed_schema_path.scheme == "http":
+        # TODO Load schema from URL
+        raise NotImplementedError("Validation from remote schema not yet supported")
+    else:
+        with open(Path(schema_path), "r") as f:
+            schema_from_path = json.load(f)
+
+    # Load data as dict
+    if isinstance(data_dict_or_path, Dict):
+        data_to_validate = data_dict_or_path
+    elif isinstance(data_dict_or_path, str):
+        with open(Path(data_dict_or_path), "r") as f:
+            data_to_validate = json.load(f)
+    else:
+        raise ValueError("Invalid data format")
+
+    # Validate data against schema
+    print(msg)
+    try:
+        jsonschema.validate(data_to_validate, schema_from_path)
+    except jsonschema.ValidationError as err:
+        # TODO Handle jsonschema.SchemaError?
```

Comment:
so you can extend error stacks in python, so I was thinking something like

```
except jsonschema.ValidationError as err:
    if msg or indent:
         nice_message = format_the_error_for_sane_people(err, msg, indent) 
         raise Exception(nice_message) from err
    raise err
```

So with that "from err" approach the jsonschema error is still in there (the "this error is cause by another error" thing you see in stack traces), we've just put a more readable exception in front of it in the trace to clue in poeple who might not get on with a raw jsonschema exception.

Worth having a play with "from err", handy trick.

I wasnt thinking anything clever for the wrapped exception literally:

```
Excaption: {msg}

{that nasty json output from jsonscema, but indented for readibility}

```

If you can do some clever field manipulation to make it more obvious then cool, but having it pretty printed rather that a block of unformatted json will do.

Reply:
Have rearranged this, but I think `if msg or indent` will always be True, as I had to specify default values in the function signature, and `data_path` and `data_dict` default to None (otherwise I get a `Non-default argument follows default argument` error)